volatility/framework/symbols/mac/__init__.py

```diff
@@ -169,9 +171,30 @@ def walk_tailq(cls,
             if len(seen) == max_elements:
                 break
 
-            yield current
+            if current.is_readable():
+                yield current
 
             try:
-                current = current.member(attr = next_member).tqe_next
+                current = current.member(attr = next_member).member(attr = list_next_member)
             except exceptions.InvalidAddressException:
                 break
+
+    @classmethod
+    def walk_tailq(cls,
+                   queue: interfaces.objects.ObjectInterface,
+                   next_member: str,
+                   max_elements: int = 4096) -> Iterable[interfaces.objects.ObjectInterface]:
+ 
+        for element in cls._walk_iterable(queue, "tqh_first", "tqe_next", next_member, max_elements):
+            yield element
+
+    @classmethod
+    def walk_list_head(cls,
```

Comment:
Actually, I forgot this is an additive change and MacUtilities is versioned, so you'll need to bump the minor version to `(1, 1, 0)` on line 34, and the plugin that uses the `walk_list_head` (ie, the PsList plugin) needs to have a dependency on MacUtilities added (that needs `(1, 1, 0)` too...